src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String STANDART_FIGURE = "Circle";
+    private static final String STANDART_COLOR = "White";
+    private static final int AMOUNT_OF_FIGURES = 4;
+    private static final int STANDART_RADIUS = 10;
+    private final ColorSupplier color;
+    private final Random random;
+    private final DimesionSupplier dimesion = new DimesionSupplier();
+    private Figure figure = null;
+
+    public FigureSupplier(ColorSupplier color, Random random) {
+        this.color = color;
+        this.random = random;
+    }
+
+    public Figure getRandomFigure() {
+        switch (Figures.values()[new Random().nextInt(Figures.values().length)]) {
+            case RECTANGLE:
+                figure = new Rectangle("Rectangle", color.getRandomColour(),
```

Comment:
you don't need the figure object, just return the object
```suggestion
               return new Rectangle("Rectangle", color.getRandomColour(),
```

Reply:
same for all